### PR TITLE
[JS] add information on custom JWT

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## What Has Changed?

Relates to: https://github.com/SAP/cloud-sdk-backlog/issues/546

Add small info on custom JWTs. While debugging an issue I saw a lot of errors in my dev tools that the class is not a dom element and One should use `className`.

### CODE REVIEW

File: docs/java/features/multi-tenancy/thread-context.mdx [hunk 1]
```diff
@@ -154,7 +154,7 @@ Assume your application received an HTTP request that contains the following hea
 - `Authorization: Bearer DUMMY_JWT`
 - `Set-Cookie: cookie-1; cookie-2`
 - `Accept-Language: en-US`
-- `x-app-specific-header: customer-value` 
+- `x-app-specific-header: customer-value`
```
Comment (marikaner) [inline docs/java/features/multi-tenancy/thread-context.mdx:157]: [req] typo?
```suggestion
- `x-app-specific-header: custome-value`
```

File: docs/java/release-notes.mdx [hunk 2]
```diff
@@ -63,7 +63,7 @@ To continue using the latest features outlined in the release notes for version
 
 - To fix the high vulnerability [CVE-2022-22968](https://nvd.nist.gov/vuln/detail/CVE-2022-22968) in Spring, we had to update [Spring BOM](https://search.maven.org/search?q=g:org.springframework%2BAND%2Ba:spring-framework-bom) to `5.3.19` and to maintain compatibility had to also update [XSUAA Client Library](https://github.com/SAP/cloud-security-xsuaa-integration) to `2.11.16`.
   Please note that if you are using SAP JAVA buildpack for your deployments use a static buildpack version `sap_java_buildpack_1_52` to avoid dependency conflicts and class loading issues.
--  SAP Buisiness Rules OpenAPI client library received breaking change to the field `List<TextPredefinedResults> predefinedResults` which is renamed to `List<TextPredefined> predefined` after updating to the latest version of the SAP Business Rules service
+- SAP Buisiness Rules OpenAPI client library received breaking change to the field `List<TextPredefinedResults> predefinedResults` which is renamed to `List<TextPredefined> predefined` after updating to the latest version of the SAP Business Rules service
```
Comment (marikaner) [inline docs/java/release-notes.mdx:66]: [req] typo

```suggestion
- SAP Business Rules OpenAPI client library received breaking change to the field `List<TextPredefinedResults> predefinedResults` which is renamed to `List<TextPredefined> predefined` after updating to the latest version of the SAP Business Rules service
```

File: docs/js/guides/how-to-retrieve-jwt.mdx [hunk 3]
```diff
@@ -37,6 +37,17 @@ The flow is as follows:
 - XSUAA issues a JWT - see [here](## Obtain JWT programmatically) for details
 - approuter adds the JWT to the request headers and redirects the request to initially requested resource
 
+A JWT issued this way contains a `JKU` header property.
+This property points to the URL where you can obtain the certificate to verify the JWT.
+The URL must be from the XSUAA domain so that it is trusted and the JWT validation does not fail.
+
+In some cases you want to use a JWT which is not issued by the XSUAA and approuter.
```
Comment (marikaner) [inline docs/js/guides/how-to-retrieve-jwt.mdx:44]: [nth] Vale suggestion
```suggestion
Sometimes you want to use a JWT which is not issued by the XSUAA and approuter.
```

File: docs/js/guides/how-to-retrieve-jwt.mdx [hunk 4]
```diff
@@ -37,6 +37,17 @@ The flow is as follows:
...
+Such tokens do not contain a `JKU` at all or a value with a different domain.
+In this case the SAP Cloud SDK will not do a validation but the destination service does.
```
Comment (marikaner) [inline docs/js/guides/how-to-retrieve-jwt.mdx:46]: [nth] Maye make it more active.
```suggestion
In this case the SAP Cloud SDK does not validate the token, but the destination service does.
```

File: docs/js/guides/how-to-retrieve-jwt.mdx [hunk 5]
```diff
@@ -37,6 +37,17 @@ The flow is as follows:
...
+You have to set the [destination properties](https://help.sap.com/docs/CP_CONNECTIVITY/cca91383641e40ffbe03bdc78f00f681/283cd2d1c72147a18c69daf681650f07.html) `x_user_token.jwks` or `x_user_token.jwks_uri` to provide the information to verify the custom JWT.
+
+If you use such a custom JWTs without the approuter and XSUAA, it is your responsibility to do the access control on the application.
```
Comment (marikaner) [inline docs/js/guides/how-to-retrieve-jwt.mdx:49]: [req] typo
[q] Should access control be specified in more detail?
```suggestion
If you use a custom JWT without the approuter and XSUAA service, it is your responsibility to do the access control in the application.
```

File: docs/js/features/connectivity/destination.mdx [hunk 6]
```diff
@@ -418,6 +418,16 @@ If you want to make HTTPS requests against systems that use certificates from ot
 For additional information on trust configuration have a look at the more [detailed guide](../../guides/trust-and-key-stores.mdx).
 Please use the `TrustAll` with great caution since it opens the gate to man in the middle attacks.
 
+### JWT Validation
+
+If you use JWTs not issued by the XSUAA you can configure validation by the destination service using the `x_user_token.jwks` or `x_user_token.jwks_uri` property.
```
Comment (marikaner) [inline docs/js/features/connectivity/destination.mdx:423]: [req] Isn't it "XSUAA service"?
```suggestion
If you use JWTs not issued by the XSUAA service you can configure validation by the destination service using the `x_user_token.jwks` or `x_user_token.jwks_uri` property.
```

File: docs/js/features/connectivity/destination.mdx [hunk 7]
```diff
@@ -418,6 +418,16 @@ If you want to make HTTPS requests against systems that use certificates from ot
...
+For more details on the JWTs have a look at the more detailed guide(../../guides/how-to-retrieve-jwt.mdx).
```
Comment (marikaner) [inline docs/js/features/connectivity/destination.mdx:424]: [req] Fix link + "the" is not necessary
```suggestion
For more details on JWTs have a look at the more detailed [guide](../../guides/how-to-retrieve-jwt.mdx).
```

**Review by marikaner — CHANGES_REQUESTED** [review]
Why are there changes in the Java docs?

File: docs/java/features/multi-tenancy/thread-context.mdx (hunk 1 above)
Comment (FrankEssenberger) [reply docs/java/features/multi-tenancy/thread-context.mdx:157]: I guess Java does not run the prettier all the time.

Comment (FrankEssenberger) [reply docs/java/features/multi-tenancy/thread-context.mdx:157]: I guess the change was introduced by the prettier. I am not sure about your suggestion what is `custome` mean?

**Review by tomfrenken — APPROVED** [review]
LGTM but I guess Marika has to re-review @marikaner

File: docs/js/features/connectivity/destination.mdx [hunk 8]
```diff
@@ -404,7 +404,7 @@ When this property is set, it is used as the header parameter `sap-client` with
 <img
   alt="SAP client property on destination"
   src={useBaseUrl('img/destination-property-sap-client.png')}
-  class="center"
+  className="center"
```
Comment (marikaner) [inline docs/js/features/connectivity/destination.mdx:407]: [q] Are you sure?

File: docs/js/features/connectivity/destination.mdx [hunk 9]
```diff
@@ -431,7 +441,7 @@ Replace `<header-key>` and `<query-key>` with the name of your headers or query
 <img
   alt="Additional properties on destination"
   src={useBaseUrl('img/destination-additional-properties.png')}
-  class="center"
+  className="center"
```
Comment (marikaner) [inline docs/js/features/connectivity/destination.mdx:444]: [q] Also, seems incorrect to me...

**Review by marikaner — APPROVED** [review]
LGTM, but please make sure "className" is correct. Is this a react construct?

**Pull request merged**